src/utils.ts

```diff
@@ -4,7 +4,28 @@ import type {
   CommonBlockProps,
   FileBlockProps,
   FolderBlockProps,
-} from "@utils";
+} from "../utils/types";
+
+import { Endpoints, OctokitResponse, RequestParameters } from "@octokit/types";
+import { endpoint } from "@octokit/endpoint";
+
+export async function onRequestGitHubData<T extends keyof Endpoints>(
+  route: T,
+  parameters?: Endpoints[T]["parameters"] & RequestParameters
+): Promise<OctokitResponse<Endpoints[T]>["data"]["response"]["data"]> {
+  const params = endpoint<T, Endpoints[T]["parameters"]>(route, parameters);
+
+  if (params.method !== "GET") {
+    throw new Error("Only GET requests are supported.");
+  }
```

Comment:
Not sure if this logic should live in `blocks-platform` or here. Or both?

Reply:
both — layered defense!